Makefile

```diff
@@ -95,16 +105,32 @@ PHONY: all pre clean fclean re test fulltest testclean testfclean retest
 
 ########################################
 
-TANIA_ENDPOINT = sources/tanya_main.c
-TANIA_ENDPOINT_OBJ = $(TANIA_ENDPOINT:.c=.o)
+TANIA_ENDPOINT_NAME = tania_main.c
+TANIA_ENDPOINT_SRCS = $(addprefix $(SOURCE_F)/,$(TANIA_ENDPOINT_NAME))
+
+TANIA_OBJ_F = build/tania/
 
-$(TANIA_ENDPOINT_OBJ): %.o: %.c
+TANIA_ENDPOINT_OBJ = $(TANIA_OBJ_F)$(TANIA_ENDPOINT_NAME:.c=.o)
+
+$(TANIA_ENDPOINT_OBJ): $(TANIA_ENDPOINT_NAME)
 	$(PREFIX)$(CC) $(COMPILE_FLAGS) $< -o $@ $(INCLUDES)
 
-tania: $(TANIA_ENDPOINT_OBJ) $(OBJS)
-	$(PREFIX)$(CC) $^ -o $(NAME) $(LINK_FLAGS)
+tania: $(OBJ_DIRS) $(OBJS) $(TANIA_ENDPOINT_OBJ)
+	$(PREFIX)$(CC) $(OBJS) $(TANIA_ENDPOINT_OBJ) -o $(NAME) $(LINK_FLAGS)
 
 ########################################
 
-vania: MOCK_FLAG += -DMOCK_TANIA
-vania: all
+VANIA_ENDPOINT_NAME = vania_main.c
+VANIA_ENDPOINT_SRCS = $(addprefix $(SOURCE_F)/,$(VANIA_ENDPOINT_NAME))
+
+VANIA_OBJ_F = build/vania/
+
+VANIA_ENDPOINT_OBJ = $(VANIA_OBJ_F)$(VANIA_ENDPOINT_NAME:.c=.o)
+
+$(VANIA_ENDPOINT_OBJ): $(VANIA_ENDPOINT_NAME)
+	$(PREFIX)$(CC) $(COMPILE_FLAGS) $< -o $@ $(INCLUDES)
+
+vania: $(OBJ_DIRS) $(OBJS) $(VANIA_ENDPOINT_OBJ)
+	$(PREFIX)$(CC) $(OBJS) $(VANIA_ENDPOINT_OBJ) -o $(NAME) $(LINK_FLAGS)
+
+vania: MOCK_FLAG += -DMOCK_TANIA
```

Comment:
Please do `checkout dev`, try `make vania` and please recreate the behavior in your PR. Also please do as little code adjustments as necessary. I need that previous version for my work. It is important for me to avoid code duplication, so I preferred to reuse `all` and not create a separate rule, and this is an important design choice. In my current stage I don't need a separate main. Please make sure you're not breaking your teammates' workflows when introducing your changes.